README.md

```diff
@@ -1,4 +1,94 @@
+# Temporal .NET SDK
 
 ⚠️ UNDER ACTIVE DEVELOPMENT
 
-(for the previous .NET SDK repo, see https://github.com/temporalio/experiment-dotnet)
\ No newline at end of file
+(for the previous .NET SDK repo, see https://github.com/temporalio/experiment-dotnet)
+
+## Quick Start
+
+TODO
+
+## Usage
+
+TODO
+
+## Development
+
+### Build
+
+With `dotnet` installed with all needed frameworks and Rust installed (i.e.
+`cargo` on the `PATH`), run:
+
+    dotnet build
+
+Or for release:
+
+    dotnet build --configuration Release
+
+### Code formatting
+
+Install [CSharpier](https://csharpier.com):
+
+    dotnet tool install --global csharpier
+
+Run `dotnet format` then `csharpier`:
+
+    dotnet format style && dotnet format analyzers && dotnet csharpier .
+
+### Testing
+
+Run:
+
+    dotnet test
+
+Can add options like:
+
+* `--logger "console;verbosity=detailed"` to show logs
+  * TODO(cretz): This doesn't show Rust stdout. How do I do that?
+* `--filter "FullyQualifiedName=Temporalio.Tests.Client.TemporalClientTests.ConnectAsync_Connection_Succeeds"` to run a
+  specific test
+
+### Rebuilding Rust extension and interop layer
+
+To rebuild DLL, make sure `protoc` is on the `PATH`, then:
```

Comment:
Clarification: Is this (`cargo build` and ClandSharpPInvokeGenerator) something that devs are expected to do manually from time to time, or will that be included in msbuild build process? Seems to me like `cargo build` is already covered by Bridge/Temporalio.csproj.

Reply:
Depending on your answer, might want to consider adding `include/temporal-sdk-bridge.h` and `Interop/*.cs` to gitignore.

Reply:
> Is this (cargo build and ClandSharpPInvokeGenerator) something that devs are expected to do manually from time to time, or will that be included in msbuild build process? Seems to me like cargo build is already covered by Bridge/Temporalio.csproj.

The `cargo build` is now part of the msbuild process. Just added, forgot to remove this from the README (this README is almost just like local dev notes for now).

The `ClandSharpPInvokeGenerator` will be expected to be invoked manually whenever any public-facing C FFI piece changes. There will be a CI step ensuring that what's in the header file matches what would be generated when re-run. We do want the generated code committed so anyone can just clone and build.